spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/DebugAddress.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2020-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.deployer.spi.local;
+
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.util.StringUtils;
+
+/**
+ * Helper for parsing the Debugging address form the legacy debug-port or the new debug-address properties.
+ * The debug-address property has precedence over debug-port and offers support for jdk 8 and jdk 9+.
+ *
+ * @author Christian Tzolov
+ */
+public class DebugAddress {
+	private static final Pattern HOSTNAME_PATTERN = Pattern.compile("^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\\-]*[A-Za-z0-9])$");
+	private static final Pattern IP_PATTERN = Pattern.compile("^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$");
+	private static final Pattern PORT_PATTERN = Pattern.compile("^([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$");
+
+	public final static Logger logger = LoggerFactory.getLogger(DebugAddress.class);
+
+	private final String host;
+	private final String port;
+	private final String address;
+	private final String suspend;
+
+	private DebugAddress(String host, int port, String suspend) {
+		this.host = host;
+		this.port = "" + port;
+		this.suspend = (StringUtils.hasText(suspend)) ? suspend.trim() : "y";
+		this.address = (StringUtils.hasText(host)) ? String.format("%s:%s", host, port) : this.port;
+	}
+
+	public String getHost() {
+		return host;
+	}
+
+	public String getPort() {
+		return port;
+	}
+
+	public String getSuspend() {
+		return suspend;
+	}
+
+	public String getAddress() {
+		return this.address;
+	}
+
+	public String getDebugCommand() {
```

Comment:
I would remove this method from this class. Single responsibility principle.

Reply:
I will move it as a default method: `getJdwpOptions(suspend, address)` inside the CommandBuilder interface.